src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -102,6 +102,14 @@ def terminate
     Puppet::Server::Config.terminate_puppet_server
   end
 
+  def getTasks(env)
+    environment = @env_loader.get(env)
+    unless environment.nil?
+      tasks_per_env =
+        Puppet::InfoService.tasks_per_environment(environment.name)
+    end
```

Comment:
you're right the assignment isn't needed. that came from some previous implementations, will remove.

And yeah, I find it weird that returns nil if environment isn't found, but that's consistent with getClassInfoForEnvironment and other methods. The code overall seems to handle this by checking the environment exists before passing requests to these methods. I'm not sure I'm a fan, but consistency seemed better unless I was going to change this everywhere? I'm very open to other thoughts on this though; going to leave as-is until I get maintainer feedback.